### DISCUSSION TITLE
ssh key does not work

### DISCUSSION THREAD

**Discussion by ZhouHuang** [community/community#35868] [Other Feature Feedback, Questions, & Ideas]
Hi everyone. Last week I noticed that my `git pull` and `git push` commands did not work. It always prompted the warning, as the following figure shows. 
<img width="808" alt="image" src="https://user-images.githubusercontent.com/23243664/195260095-c40f20c0-c893-43c7-9b79-8aa8b62cccc3.png">
Well I deleted the git@github.com line in known_hosts. The warning disapears, however, still does not work for me. The command `git pull` or `git push` requeries the password from the keyboard, as the following figure shows.
<img width="785" alt="image" src="https://user-images.githubusercontent.com/23243664/195260531-2c523e69-6616-419a-9372-770a24827f4a.png">
It's quite strange as my ssh key does not change after 2020 at all.
The debug infomation is shown in the following.
```
▶ ssh -v git@github.com
OpenSSH_8.1p1, LibreSSL 2.7.3
debug1: Reading configuration data /Users/zhou/.ssh/config
debug1: Reading configuration data /etc/ssh/ssh_config
debug1: /etc/ssh/ssh_config line 47: Applying options for *
debug1: /etc/ssh/ssh_config line 51: Applying options for *
debug1: Connecting to github.com port 22.
debug1: Connection established.
debug1: identity file /Users/zhou/.ssh/id_rsa type 0
debug1: identity file /Users/zhou/.ssh/id_rsa-cert type -1
debug1: identity file /Users/zhou/.ssh/id_dsa type -1
debug1: identity file /Users/zhou/.ssh/id_dsa-cert type -1
debug1: identity file /Users/zhou/.ssh/id_ecdsa type -1
debug1: identity file /Users/zhou/.ssh/id_ecdsa-cert type -1
debug1: identity file /Users/zhou/.ssh/id_ed25519 type -1
debug1: identity file /Users/zhou/.ssh/id_ed25519-cert type -1
debug1: identity file /Users/zhou/.ssh/id_xmss type -1
debug1: identity file /Users/zhou/.ssh/id_xmss-cert type -1
debug1: Local version string SSH-2.0-OpenSSH_8.1
debug1: Remote protocol version 2.0, remote software version OpenSSH_7.6
debug1: match: OpenSSH_7.6 pat OpenSSH_7.0*,OpenSSH_7.1*,OpenSSH_7.2*,OpenSSH_7.3*,OpenSSH_7.4*,OpenSSH_7.5*,OpenSSH_7.6*,OpenSSH_7.7* compat 0x04000002
debug1: Authenticating to github.com:22 as 'git'
debug1: SSH2_MSG_KEXINIT sent
debug1: SSH2_MSG_KEXINIT received
debug1: kex: algorithm: curve25519-sha256
debug1: kex: host key algorithm: rsa-sha2-512
debug1: kex: server->client cipher: chacha20-poly1305@openssh.com MAC: <implicit> compression: none
debug1: kex: client->server cipher: chacha20-poly1305@openssh.com MAC: <implicit> compression: none
debug1: expecting SSH2_MSG_KEX_ECDH_REPLY
debug1: Server host key: ssh-rsa SHA256:rEmlJenVMSL5GVemSY0Gk8WGw6B4ege4J85M+vup8R0
debug1: Host 'github.com' is known and matches the RSA host key.
debug1: Found key in /Users/zhou/.ssh/known_hosts:23
debug1: rekey out after 134217728 blocks
debug1: SSH2_MSG_NEWKEYS sent
debug1: expecting SSH2_MSG_NEWKEYS
debug1: SSH2_MSG_NEWKEYS received
debug1: rekey in after 134217728 blocks
debug1: Will attempt key: /Users/zhou/.ssh/id_rsa RSA SHA256:oRTX/C4a8fEQqjgOcQRY9QaMhih90KbxHA1PxzbEl08
debug1: Will attempt key: /Users/zhou/.ssh/id_dsa
debug1: Will attempt key: /Users/zhou/.ssh/id_ecdsa
debug1: Will attempt key: /Users/zhou/.ssh/id_ed25519
debug1: Will attempt key: /Users/zhou/.ssh/id_xmss
debug1: SSH2_MSG_EXT_INFO received
debug1: kex_input_ext_info: server-sig-algs=<ssh-ed25519,ssh-rsa,rsa-sha2-256,rsa-sha2-512,ssh-dss>
debug1: SSH2_MSG_SERVICE_ACCEPT received
debug1: Authentications that can continue: publickey,password,keyboard-interactive
debug1: Next authentication method: publickey
debug1: Offering public key: /Users/zhou/.ssh/id_rsa RSA SHA256:oRTX/C4a8fEQqjgOcQRY9QaMhih90KbxHA1PxzbEl08
debug1: Authentications that can continue: publickey,password,keyboard-interactive
debug1: Trying private key: /Users/zhou/.ssh/id_dsa
debug1: Trying private key: /Users/zhou/.ssh/id_ecdsa
debug1: Trying private key: /Users/zhou/.ssh/id_ed25519
debug1: Trying private key: /Users/zhou/.ssh/id_xmss
debug1: Next authentication method: keyboard-interactive
debug1: Authentications that can continue: publickey,password,keyboard-interactive
debug1: Next authentication method: password
git@github.com's password:
```

Wonder anyone could help? Thanks a lot.

**Comment by airtower-luna** [comment]
Removing the line for `github.com` in `known_hosts` is an extremely dangerous step, it basically means you chose to ignore the warning that the remote server is unknown. The screenshot shows that the key fingerprint of whatever server you're connecting to *isn't one of [GitHub's published public keys](https://docs.github.com/en/authentication/keeping-your-account-and-data-secure/githubs-ssh-key-fingerprints)*.

That means someone or something is intercepting your connection. Whatever server you're connecting to isn't GitHub. I can't say who, what, or why, but you should investigate that instead of ignoring the warning. Could be anything from a misconfiguration somewhere to spies, really. :sweat_smile: But if you ignore it, your data will go to whoever is messing with your connection.

**Reply by ZhouHuang** [reply]
yeah, same consideration... Could't believe I was hacked in some ways. Thanks for your reply anyway!

**Reply by airtower-luna** [reply]
Nothing here implies you were hacked, someone is messing with your network connection. Getting into your computer may or may not be part of their goal.

I notice the exact same public key has been mentioned in #35917. You might want to look for similarities, maybe you're both in the same network at some level? E.g. same university/company (in that case I'd suspect a misconfigured or hacked router/firewall), provider, or country (maybe an attempt at censorship/surveillance)?

**Comment by iamgauravpant** [comment]
If you're using Windows OS try this : https://www.youtube.com/watch?v=WgZIv5HI44o

**Reply by ZhouHuang** [reply]
I am using MacOS. Thanks for your information!

**Comment by VNemo** [comment]


**Reply by ZhouHuang** [reply]
可能是开会原因吧。还好我没有太多重要的repo在github上。

**Comment by Haolun123** [comment]
Yep, same macOS, same issue. Pretty strange... And no clue why it's happening. No solution at all. Also looking for help...

**Reply by jiaohu** [reply]
did you solve it ?😂

**Comment by jiaohu** [comment]
> Hi everyone. Last week I noticed that my `git pull` and `git push` commands did not work. It always prompted the warning, as the following figure shows. <img alt="image" width="808" src="https://user-images.githubusercontent.com/23243664/195260095-c40f20c0-c893-43c7-9b79-8aa8b62cccc3.png"> Well I deleted the [git@github.com](mailto:git@github.com) line in known_hosts. The warning disapears, however, still does not work for me. The command `git pull` or `git push` requeries the password from the keyboard, as the following figure shows. <img alt="image" width="785" src="https://user-images.githubusercontent.com/23243664/195260531-2c523e69-6616-419a-9372-770a24827f4a.png"> It's quite strange as my ssh key does not change after 2020 at all. The debug infomation is shown in the following.
> 
> ```
> ▶ ssh -v git@github.com
> OpenSSH_8.1p1, LibreSSL 2.7.3
> debug1: Reading configuration data /Users/zhou/.ssh/config
> debug1: Reading configuration data /etc/ssh/ssh_config
> debug1: /etc/ssh/ssh_config line 47: Applying options for *
> debug1: /etc/ssh/ssh_config line 51: Applying options for *
> debug1: Connecting to github.com port 22.
> debug1: Connection established.
> debug1: identity file /Users/zhou/.ssh/id_rsa type 0
> debug1: identity file /Users/zhou/.ssh/id_rsa-cert type -1
> debug1: identity file /Users/zhou/.ssh/id_dsa type -1
> debug1: identity file /Users/zhou/.ssh/id_dsa-cert type -1
> debug1: identity file /Users/zhou/.ssh/id_ecdsa type -1
> debug1: identity file /Users/zhou/.ssh/id_ecdsa-cert type -1
> debug1: identity file /Users/zhou/.ssh/id_ed25519 type -1
> debug1: identity file /Users/zhou/.ssh/id_ed25519-cert type -1
> debug1: identity file /Users/zhou/.ssh/id_xmss type -1
> debug1: identity file /Users/zhou/.ssh/id_xmss-cert type -1
> debug1: Local version string SSH-2.0-OpenSSH_8.1
> debug1: Remote protocol version 2.0, remote software version OpenSSH_7.6
> debug1: match: OpenSSH_7.6 pat OpenSSH_7.0*,OpenSSH_7.1*,OpenSSH_7.2*,OpenSSH_7.3*,OpenSSH_7.4*,OpenSSH_7.5*,OpenSSH_7.6*,OpenSSH_7.7* compat 0x04000002
> debug1: Authenticating to github.com:22 as 'git'
> debug1: SSH2_MSG_KEXINIT sent
> debug1: SSH2_MSG_KEXINIT received
> debug1: kex: algorithm: curve25519-sha256
> debug1: kex: host key algorithm: rsa-sha2-512
> debug1: kex: server->client cipher: chacha20-poly1305@openssh.com MAC: <implicit> compression: none
> debug1: kex: client->server cipher: chacha20-poly1305@openssh.com MAC: <implicit> compression: none
> debug1: expecting SSH2_MSG_KEX_ECDH_REPLY
> debug1: Server host key: ssh-rsa SHA256:rEmlJenVMSL5GVemSY0Gk8WGw6B4ege4J85M+vup8R0
> debug1: Host 'github.com' is known and matches the RSA host key.
> debug1: Found key in /Users/zhou/.ssh/known_hosts:23
> debug1: rekey out after 134217728 blocks
> debug1: SSH2_MSG_NEWKEYS sent
> debug1: expecting SSH2_MSG_NEWKEYS
> debug1: SSH2_MSG_NEWKEYS received
> debug1: rekey in after 134217728 blocks
> debug1: Will attempt key: /Users/zhou/.ssh/id_rsa RSA SHA256:oRTX/C4a8fEQqjgOcQRY9QaMhih90KbxHA1PxzbEl08
> debug1: Will attempt key: /Users/zhou/.ssh/id_dsa
> debug1: Will attempt key: /Users/zhou/.ssh/id_ecdsa
> debug1: Will attempt key: /Users/zhou/.ssh/id_ed25519
> debug1: Will attempt key: /Users/zhou/.ssh/id_xmss
> debug1: SSH2_MSG_EXT_INFO received
> debug1: kex_input_ext_info: server-sig-algs=<ssh-ed25519,ssh-rsa,rsa-sha2-256,rsa-sha2-512,ssh-dss>
> debug1: SSH2_MSG_SERVICE_ACCEPT received
> debug1: Authentications that can continue: publickey,password,keyboard-interactive
> debug1: Next authentication method: publickey
> debug1: Offering public key: /Users/zhou/.ssh/id_rsa RSA SHA256:oRTX/C4a8fEQqjgOcQRY9QaMhih90KbxHA1PxzbEl08
> debug1: Authentications that can continue: publickey,password,keyboard-interactive
> debug1: Trying private key: /Users/zhou/.ssh/id_dsa
> debug1: Trying private key: /Users/zhou/.ssh/id_ecdsa
> debug1: Trying private key: /Users/zhou/.ssh/id_ed25519
> debug1: Trying private key: /Users/zhou/.ssh/id_xmss
> debug1: Next authentication method: keyboard-interactive
> debug1: Authentications that can continue: publickey,password,keyboard-interactive
> debug1: Next authentication method: password
> git@github.com's password:
> ```
> 
> Wonder anyone could help? Thanks a lot.

did you solve the problem? I meet it too😭😭😭

**Reply by brunolinux** [reply]
you can try this: https://www.v2ex.com/t/881922
